pkg/scheduler/controller/shoot/distance_test.go

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package shoot
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Scheduler_Control", func() {
+	Context("orientation", func() {
+		It("handles name without orientation", func() {
+			base, orient := orientation("europe")
+			Expect(base).To(Equal("europe"))
+			Expect(orient).To(Equal(""))
+		})
+
+		It("handles orientation prefix", func() {
+			base, orient := orientation("west_europe")
+			Expect(base).To(Equal(":_europe"))
+			Expect(orient).To(Equal("west"))
+		})
+
+		It("handles orientation suffix", func() {
+			base, orient := orientation("europe-west")
+			Expect(base).To(Equal("europe-:"))
+			Expect(orient).To(Equal("west"))
+		})
+		It("handles orientation in the middle", func() {
+			base, orient := orientation("middle-west-europe")
+			Expect(base).To(Equal("middle-:-europe"))
+			Expect(orient).To(Equal("west"))
+		})
+	})
+
+	Context("orientation", func() {
```

Comment:
There is already a context with the same name above. Probably you wanted to name it `distance`?

```suggestion
	Context("distance", func() {
```